### DISCUSSION TITLE
ESP32-WROOM-32E  (BLE + WIFI) Firmware

### DISCUSSION THREAD

**Discussion by AlexAlvarezAlmendros** [nanoframework/Home#1342] [Q&A] [answered]
I need to use BluetoothLE and Wifi at once to scan wifi ssid's and send them to my app with bluetooth. I have been testing it but I need to change the firmware every time. For Wifi I need ESP32_REV3 and for BLE I need ESP32_BLE_REV3. I want to know if it exists a firmware to do both Wifi and BLE at once.

**Comment by alberk8** [comment, marked as answer]
There is one firmware EPS32_BLE_REV3 and it can do both.   Use the command `nanoff --target ESP32_BLE_REV3 --update --serialport COM3`
If you cannot most likely it is due to insufficient memory. You will need a ESP32 REV3 with PSRAM.   I can verify that both Wifi and BLE can work concurrent with ESP32 REV3 with PSRAM (ESP32 WROVER-E)

**Reply by AlexAlvarezAlmendros** [reply]
Okey, could you verify if they can work together at the same time? It will be awesome!

**Reply by alberk8** [reply]
Yes, they can work simultaneously.  Wifi is UP and connect to a MQTT server and BLE is scanning the background and connect to device with a specific MAC address, connects to it, get data and disconnect on a periodic basis.  Once data from BLE is obtained it is set to server via MQTT connection.

**Reply by AlexAlvarezAlmendros** [reply]
thx
